mesh/accounts/services.py

```diff
@@ -0,0 +1,80 @@
+from .models import Account
+from django.http import JsonResponse
+from django.core.mail import send_mail
+import pyotp, bcrypt, json, os
+
+def decrypt(password, salt):
+    pepper = os.getenv("PEPPER")
+    password = f"{password}{pepper}".encode('utf-8')
+    return bcrypt.hashpw(password,salt)
+
+def getUserServices(request):
+    """
+    Get the account object
+    
+    @return: account object if id exist, None otherwise
+    """
+    try:
+        data = json.loads(request.body.decode('utf-8')) 
+        account_id = data.get('accountID', None)
+        account = Account.objects.get(accountID=account_id)
+        return account
+    except Account.DoesNotExist:
+        return None
+
+
+def postEmailCodeService(user):
+    """
+    Check if the user has otp seed generated
+    If it does not exist, random base32 otp seed will be generated and saved for this user
+    otherwise, use the saved otp seed
+
+    @return
+    no return value
+    """
+    if(not user.enabled2Factor):
+        #check if the user has 2fa enabled, if not generate one
+        user.enabled2Factor = True
+        otp_base32 = pyotp.random_base32()
+        user.otp_base32 = otp_base32
+        user.save()
+    else:
+        otp_base32 = user.otp_base32
+    totp = pyotp.TOTP(otp_base32, interval=60) #for debug, remove after sending email works
+    #sending email need authentication
+    # send_mail(     
+    #     "User OTP",
+    #     "the otp: {}".format(pyotp.TOTP(otp_base32, interval=90).now),
+    #     os.environ.get("EMAIL_NAME"),
+    #     [user.email],
+    #     fail_silently=False,
+    # )
+    print(totp.now())   #for debug. remove after sending email works
```

Comment:
For parts like these, if it is not being handled in this PR then an issue should be created. I saw that you mentioned it in the PR description which is good but it would be more clear if there were issues, and I would write in those issues what should be removed just in case people don't stumble across these comments.

Reply:
Should the issue be made before or after the PR is merged? I was assuming that since the changes haven't been approved yet I shouldn't make any issue if it's related to in-progress changes.

Reply:
You're correct, it probably should be made after your work is merged. Link this PR to the issues you may create so that people can read the description to know what they should be looking to achieve.